test/integration/sys-mac.int.c

```diff
@@ -0,0 +1,153 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*******************************************************************************
+ * Copyright 2017-2018, Fraunhofer SIT sponsored by Infineon Technologies AG
+ * All rights reserved.
+ *******************************************************************************/
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include "tss2_tpm2_types.h"
+#include "tss2_sys.h"
+#include "util/tpm2b.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <inttypes.h>
+
+#include "tss2_tcti.h"
+#include "tss2_sys.h"
+
+#define LOGMODULE test
+#include "util/log.h"
+#include "test.h"
+#include "sys-util.h"
+
+int
+test_sys_mac(TSS2_SYS_CONTEXT * sys_context)
+{
+
+    TSS2_RC rc;
+    TPM2B_PUBLIC out_public = { 0 };
+    TPM2B_CREATION_DATA creation_data = {0,};
+    creation_data.size = 0;
+    TPM2B_DIGEST creation_hash = {
+        .size = sizeof(creation_hash.buffer)
+    };
+    TPMT_TK_CREATION creation_ticket = { 0 };
+    TPM2B_NAME name;
+    /* session parameters */
+    /* command session info */
+    TSS2L_SYS_AUTH_COMMAND  sessions_cmd = {
+        .auths = {{ .sessionHandle = TPM2_RH_PW }},
+        .count = 1
```

Comment:
@malikabhi05 Let me explain this data structure. A TPM command can have 3 sessions for various purposes. In the command spec, it will show the location of the session for authorizing usage of the object/key. For instance, if we look at TPM2_MAC command, it shows in Table 66 that @handle has an Auth Index of 1. This means the first thing in the array of sessions is a session that can be used for authorizing the object. In this case, we're going to use the password session value of TPM2_RH_PW, which means that the buffer in that index slot for the auth value has to be set into the hmac field, which is of type TPM2B_AUTH. Below we assign the primary sensitive authValue to authValuePrimary but we never set the hmac field in variable `session_cmd.auths[0].hmac`.

The SIMPLEST way to fix this is to delete authValuePrimary and just use objects without passwords, thus all you need is what's set up. This is the same for the HMAC test as well.